libs/s25main/controls/ctrlBaseTooltip.h

```diff
@@ -26,4 +26,5 @@ class ctrlBaseTooltip
 
 protected:
     std::string tooltip_;
+    mutable bool showing_ = false;
```

Comment:
please only init inside the constructor (so we have *one* place to initialize things)

I always find `mutable` ugly, but I also don't see a better way to do it in this case.

So basically it lgtm, besides the one small thing.

Thanks

Reply:
> please only init inside the constructor (so we have _one_ place to initialize things)

Will do. Maybe worth considering changing that policy to always initialize in the class body at some point. Tends to avoid mistakes, in my experience, and is generally considered best practice since C++11.

> I always find `mutable` ugly, but I also don't see a better way to do it in this case.

Dito.

Reply:
I don't mind having in-class initializers and ctors setting stuff. That allows to have defaults in-class and ctors only set base on their arguments. Especially good for classes with multiple different ctors having an attribute like this "show" think which is always off by default.